tpm2/test/hierarchy_change_auth_test.go

```diff
@@ -0,0 +1,75 @@
+package tpm2test
+
+import (
+	"testing"
+
+	. "github.com/google/go-tpm/tpm2"
+	"github.com/google/go-tpm/tpm2/transport/simulator"
+)
+
+func TestHierarchyChangeAuth(t *testing.T) {
+	thetpm, err := simulator.OpenSimulator()
+	if err != nil {
+		t.Fatalf("could not connect to TPM simulator: %v", err)
+	}
+	defer thetpm.Close()
+
+	authKey := []byte("authkey")
+	newAuthKey := []byte("newAuthKey")
+
+	t.Run("HierarchyChangeAuthOwner", func(t *testing.T) {
+		hca := HierarchyChangeAuth{
+			AuthHandle: TPMRHOwner,
+			NewAuth: TPM2BAuth{
+				Buffer: authKey,
+			},
+		}
+
+		_, err := hca.Execute(thetpm)
+		if err != nil {
+			t.Fatalf("failed HierarchyChangeAuth: %v", err)
```

Comment:
nit: Some of these `t.Fatal` cases could be `t.Error` since they aren't must-succeed setup steps. Not a big deal to me either way

Reply:
Can you clarify that? In which case would it be okay if this test fails?

Reply:
Sure! Basically, https://google.github.io/styleguide/go/decisions#keep-going:

> Tests should keep going for as long as possible, even after a failure, in order to print out all of the failed checks in a single run. This way, a developer who is fixing the failing test doesn’t have to re-run the test after fixing each bug to find the next bug.
>
> Prefer calling t.Error over t.Fatal for reporting a mismatch. When comparing several different properties of a function’s output, use t.Error for each of those comparisons.
>
> Calling t.Fatal is primarily useful for reporting an unexpected error condition, when subsequent comparison failures are not going to be meaningful.

The cases in this PR are somewhat borderline since the tests are nice and simple.